### PR TITLE
BlockGoblin now can throw spikeblock

### PR DESCRIPTION
**Pull request by ghost**
I tried to improve the attack method of block Goblin
If he sees an enemy, he sometimes throws the iron ball(Or does it say spikeball?)

### CODE REVIEW

File: src/main/java/twilightforest/entity/EntityTFBlockGoblin.java [hunk 1]
```diff
@@ -24,22 +26,31 @@
...
 
 	public static final ResourceLocation LOOT_TABLE = TwilightForestMod.prefix("entities/block_goblin");
 	private static final float CHAIN_SPEED = 16F;
 	private static final DataParameter<Byte> DATA_CHAINLENGTH = EntityDataManager.createKey(EntityTFBlockGoblin.class, DataSerializers.BYTE);
 	private static final DataParameter<Byte> DATA_CHAINPOS = EntityDataManager.createKey(EntityTFBlockGoblin.class, DataSerializers.BYTE);
+	private static final DataParameter<Boolean> ISTHROW = EntityDataManager.createKey(EntityTFBlockGoblin.class, DataSerializers.BOOLEAN);
 
 	private int recoilCounter;
 	private float chainAngle;
+	private float chainMoveLengh0;
```
Comment (Drullkus) [inline src/main/java/twilightforest/entity/EntityTFBlockGoblin.java]: Is `chainMoveLengh0` required? It looks like it's always written to and never read from. Looks redundant I think.

Also `Lengh` -> `Length`. Missing the `t`

File: src/main/java/twilightforest/entity/EntityTFBlockGoblin.java [hunk 2]
```diff
@@ -24,22 +26,31 @@
...
 public class EntityTFBlockGoblin extends EntityMob implements IEntityMultiPart {
+	private static final UUID MODIFIER_UUID = UUID.fromString("5CD17E52-A79A-43D3-A529-90FDE04B181E");
+	private static final AttributeModifier MODIFIER = (new AttributeModifier(MODIFIER_UUID, "Drinking speed penalty", -0.25D, 0)).setSaved(false);
+
 
 	public static final ResourceLocation LOOT_TABLE = TwilightForestMod.prefix("entities/block_goblin");
 	private static final float CHAIN_SPEED = 16F;
 	private static final DataParameter<Byte> DATA_CHAINLENGTH = EntityDataManager.createKey(EntityTFBlockGoblin.class, DataSerializers.BYTE);
 	private static final DataParameter<Byte> DATA_CHAINPOS = EntityDataManager.createKey(EntityTFBlockGoblin.class, DataSerializers.BYTE);
+	private static final DataParameter<Boolean> ISTHROW = EntityDataManager.createKey(EntityTFBlockGoblin.class, DataSerializers.BOOLEAN);
```
Comment (Drullkus) [inline src/main/java/twilightforest/entity/EntityTFBlockGoblin.java]: `ISTHROW` should be renamed to `IS_THROWING`

File: src/main/java/twilightforest/entity/EntityTFBlockGoblin.java [hunk 3]
```diff
@@ -216,12 +281,24 @@ protected void applyBlockCollision(Entity collider, Entity collided) {
...
 
 			}
 		}
 	}
 
+	public boolean isThrow() {
+		return this.dataManager.get(ISTHROW);
+	}
+
+	public void setThrow(boolean isthrow) {
```
Comment (Drullkus) [inline src/main/java/twilightforest/entity/EntityTFBlockGoblin.java]: `isThrow` and `setThrow` should be renamed to `isThrowing` and `setThrowing`

File: src/main/java/twilightforest/entity/EntityTFBlockGoblin.java [hunk 4]
```diff
@@ -24,22 +26,31 @@
...
 import net.minecraft.world.World;
 import twilightforest.TFSounds;
 import twilightforest.TwilightForestMod;
 
 import java.util.List;
+import java.util.UUID;
 
 public class EntityTFBlockGoblin extends EntityMob implements IEntityMultiPart {
+	private static final UUID MODIFIER_UUID = UUID.fromString("5CD17E52-A79A-43D3-A529-90FDE04B181E");
+	private static final AttributeModifier MODIFIER = (new AttributeModifier(MODIFIER_UUID, "Drinking speed penalty", -0.25D, 0)).setSaved(false);
```
Comment (Drullkus) [inline src/main/java/twilightforest/entity/EntityTFBlockGoblin.java]: `Drinking speed penalty` wouldn't accurately describe the function of the attribute modifier, I recommend renaming to just `speedPenalty` or something like that.

**Review by Drullkus — CHANGES_REQUESTED** [review]
So there's an unused `float` field `chainMoveLengh0`, and also just some naming fixes would be appreciated, but otherwise I think the code itself looks good

**Review by Drullkus — APPROVED** [review]

**Comment by Tamaized** [comment]
Could this instead be an ai task rather than shoving it all in onUpdate

**Comment by ghost** [comment]
I added a dedicated AI

**Review by Tamaized — APPROVED** [review]

**Pull request merged**